### PR TITLE
[Feature] Add array function of array_flatten

### LINKED ISSUE #19819: [function] array_flatten
- [x] https://github.com/StarRocks/starrocks/pull/50080
need a function to flatten one layer of nested arrays. 

## description

Flattens an array(array(T)) to an array(T) by concatenating the contained arrays.

## syntax

```
array_flatten(expr)
```

in which, expr is an nested array.

## examples

```
SELECT array_flatten([[1, 2], [1, 4]]);
 [1,2,1,4]

select array_flatten([[[1],[2]],[[3],[4]]]);
 [[1], [2], [3], [4]] 

```

### PR DESCRIPTION
**Pull request by sfwang218**
## What I'm doing:

Fixes #19819

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [x] This pr needs user documentation (for new or modified features or behaviors)
  - [x] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by Seaven** [comment]
Could you add more sql test case about combination of array function, like: array_slice(array_flatten)... array_flatten(array_slice)...

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -1492,6 +1492,21 @@ private void checkFunction(String fnName, FunctionCallExpr node, Type[] argument
                     }
                     break;
                 }
+                case FunctionSet.ARRAY_FLATTEN: {
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/ExpressionAnalyzer.java:1495]: add FE UT plz.

and I think you lose hanlde result type in ExpressionAnalzye？ you can find it in ExpressionAnalyze.visitFunctionCall and PolymorphicFunctionAnalyzer

**Comment by sfwang218** [comment]
> Could you add more sql test case about combination of array function, like: array_slice(array_flatten)... array_flatten(array_slice)...

I don't know how to add sql test, could you please provide me with some reference samples?

**Comment by wangsimo0** [comment]
> > Could you add more sql test case about combination of array function, like: array_slice(array_flatten)... array_flatten(array_slice)...
> 
> I don't know how to add sql test, could you please provide me with some reference samples?

you can find SQL test sample in merged function pr like this one https://github.com/StarRocks/starrocks/pull/17905

**Comment by sfwang218** [comment]
> wangsimo0



> > > Could you add more sql test case about combination of array function, like: array_slice(array_flatten)... array_flatten(array_slice)...
> > 
> > 
> > I don't know how to add sql test, could you please provide me with some reference samples?
> 
> you can find SQL test sample in merged function pr like this one #17905



> > > Could you add more sql test case about combination of array function, like: array_slice(array_flatten)... array_flatten(array_slice)...
> > 
> > 
> > I don't know how to add sql test, could you please provide me with some reference samples?
> 
> you can find SQL test sample in merged function pr like this one #17905

Hello, I've just opened a new pull request #50080 , and I will close this one. Could you please review it when you have time? Thank you.

**Pull request closed without merging**